src/main/java/ch/njol/skript/conditions/CondExplodesWithFire.java

```diff
@@ -0,0 +1,92 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright 2011-2017 Peter Güttinger and contributors
+ */
+package ch.njol.skript.conditions;
+
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Explosive;
+import org.bukkit.event.Event;
+import org.bukkit.event.entity.ExplosionPrimeEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+import ch.njol.skript.ScriptLoader;
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Condition;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.log.ErrorQuality;
+import ch.njol.util.Kleenean;
+
+@Name("Explodes With Fire")
+@Description("Checks if an entity will create fire if it explodes.")
+@Examples({"on explosion prime:", 
+			"\tif the event-explosion is fiery:",
+			"\t\tbroadcast \"A fiery explosive has been ignited!\""})
```

Comment:
Needs annotation to indicate the event it works in.

Reply:
Do you think this could confuse some users, as the syntax is usable outside of the event? I mean, reading the patterns would probably clear up confusion, but maybe it would be better to specify that the "event-explosion" pattern is only usable in the explosion prime event?

Reply:
Hmm, true. Maybe it is better to leave it out.